### ISSUE TITLE
Retain/keep dependency name even if not found or provide a function to retrieve it

### ISSUE THREAD

**Issue by mv0** [mesonbuild/meson#9809]
Trying to retrieve the dependency name using `name()` function, results in `not-found` string, when the dependency is not found. This means that there's need
to add additional data/information to store the dependency name, which is not always possible.

**To Reproduce**
```
dep = dependency('gbm', version: '>= 21.1.1', required: false)
if not dep.found()
    message('Could not find dependency ' + dep.name())
```

**Expected behavior**
Either return the dependency name or provide an alternative to `dep.name()` which retains the name for easy access.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
No
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
Debian testing
* what Python version are you using e.g. 3.8.0
Python 3.9
* what `meson --version`
0.60.2
* what `ninja --version` if it's a Ninja build
1.10.1

If this reasonable I can try to fix it myself, but would've like to get some feedback first, maybe I'm missing some feat that I can use instead?

**Comment by radioactiveman** [comment]
Duplicate of my bug report https://github.com/mesonbuild/meson/issues/8114 which has been fixed already. You need to update Meson.

**Comment by eli-schwartz** [comment]
Duplicate of #8114

**Comment by eli-schwartz** [comment]
Indeed, fixed in 0.61

**Comment by mv0** [comment]
Awesome, thanks a lot for the heads up. Good to know it's fixed.
